pkg/ipamd/datastore/prefix_store.go

```diff
@@ -0,0 +1,176 @@
+package datastore
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/utils/logger"
+	"github.com/pkg/errors"
+)
+
+const (
+	//IP tracking bitmap is stored in size of octets
+	octetSize = 8
+
+	//"/32" IPv4 prefix
+	ipv4DefaultPrefixSize = 32
+)
+
+var log = logger.Get()
+
+// PrefixIPsStore will hold the size of the prefix Eg: 16 for /28 prefix
+// UsedIPs will be a bitmap of size - IPsPerPrefix which will keep track of IPs used
+type PrefixIPsStore struct {
+	UsedIPs      []byte
+	CooldownIPs  []time.Time
+	IPsPerPrefix int
+}
+
+// ENIPrefix will have the prefixes allocated for an ENI and the list of allocated IPs under
+// the prefix
+type ENIPrefix struct {
```

Comment:
This should have a `String()` method, or (better?) have an easy way to get `Prefix`/`PrefixLen` as a `net.IPNet` (and then we could get a string from there).

Then we can remove several other places that are currently doing their own `prefix + "/" strconv.Itoa(PrefixLen)`

Reply:
Yes, I was actually working on that. Will update it.

Reply:
Fixed this.